### PR TITLE
Core: generate and parse railjson v2

### LINKED ISSUE #378: Adapt core to railjson v2
- [x] Modify railjson generator to v2
- [x] Adapt core parser to v2
- [x] Rewrite hand-written infras (tiny infra)

### PR DESCRIPTION
**Pull request by eckter**
Fixes https://github.com/DGEXSolutions/osrd/issues/378

- [x] Modify railjson generator to v2
- [x] Adapt core parser to v2
- [x] Rewrite hand-written infras (tiny infra)

### CODE REVIEW

File: core/examples/generator/railjson_generator/schema/infra/route.py [hunk 1]
```diff
@@ -19,23 +15,33 @@ class Route:
     entry_direction: Direction
     switches_group: Mapping[str, str] = field(default_factory=dict)
     tvd_sections: List[TVDSection] = field(default_factory=list)
-    label: str = field(default_factory=_route_id)
+    label: str = field(default=None)
     path_elements: List["PathElement"] = field(default_factory=list)
 
     _INDEX = 0
```
Comment (flomonster) [inline core/examples/generator/railjson_generator/schema/infra/route.py:21]: Can be removed

File: core/examples/generator/schemas.py [hunk 2]
```diff
@@ -0,0 +1 @@
+../../../api/osrd_infra/models/schemas.py
```
Comment (flomonster) [inline core/examples/generator/schemas.py]: Symlink won't work with windows... One way to fix this is to use `importlib`

File: core/src/main/java/fr/sncf/osrd/railjson/schema/infra/RJSOperationalPoint.java [hunk 3]
```diff
@@ -1,18 +1,32 @@
...
+
+    public RJSOperationalPoint(String name, List<RJSOperationalPointPart> parts) {
+        this.name = name;
+        this.parts = parts;
     }
 
     @Override
     public String getID() {
-        return id;
+        return name;
```
Comment (flomonster) [inline core/src/main/java/fr/sncf/osrd/railjson/schema/infra/RJSOperationalPoint.java]: We shouldn't use name as `ID`, the field `id` is missing and should be use as unique identifier.
I'm not sure we need the following fields: `name`, `ci`, `ch`, `chShortLabel` and `chLongLabel`.

File: core/src/main/java/fr/sncf/osrd/railjson/schema/infra/trackobjects/RJSSignal.java [hunk 4]
```diff
@@ -23,28 +24,27 @@
 
     /** Detector linked with the signal, may be empty if the signal doesn't protect a route */
     @Json(name = "linked_detector")
-    public ID<RJSTrainDetector> linkedDetector = new ID<>("");
+    public RJSObjectRef<RJSTrainDetector> linkedDetector;
+
+    @SuppressFBWarnings("URF_UNREAD_PUBLIC_OR_PROTECTED_FIELD")
+    public double angle;
```
Comment (flomonster) [inline core/src/main/java/fr/sncf/osrd/railjson/schema/infra/trackobjects/RJSSignal.java]: `angle` is not used

**Review by flomonster — CHANGES_REQUESTED** [review]
Great changes! Few core rjs schemas has fields not used by the simulator. I don't think it's necessary to specify these fields (I may have missed some of them).

File: core/examples/generator/railjson_generator/schema/infra/route.py (hunk 1 above)
Comment (eckter) [reply core/examples/generator/railjson_generator/schema/infra/route.py:21]: It doesn't seem so, removing it makes it a required field.

File: core/examples/generator/railjson_generator/__init__.py [hunk 5]
```diff
@@ -1,3 +1,9 @@
+import sys
+import os
+
+schema_path = os.path.dirname(__file__) + "/../../../../api/osrd_infra/models"
+sys.path.append(schema_path)
```
Comment (flomonster) [inline core/examples/generator/railjson_generator/__init__.py]: You can use `pathlib` instead of `os.path`

**Review by flomonster — APPROVED** [review]

**Pull request merged**